rfcs/text/0000-application-monitoring-for-ga.md

```diff
@@ -0,0 +1,146 @@
+### DCP PR:
+
+# System monitoring for General Availability
+
+## Summary
+
+This RFC describes the first standard revision for monitoring the distributed systems of the Human Cell Atlas (HCA) Data Coordination Platform (DCP). The standard is scoped to the minimum viable set of monitoring infrastructure the DCP will need to reach General Availability (GA). We anticipate that future revisions will be needed to meet our Service Level Objectives (SLOs) as DCP matures.
+
+The term monitoring is used here as defined in the [Google Site Reliability Engineering (SRE)](https://landing.google.com/sre/sre-book/chapters/monitoring-distributed-systems/) program. This document borrows heavily from SRE.
+
+## Author(s)
+
+* [Matthew Weiden](mailto:mweiden@chanzuckerberg.com)
+
+## Shepherd
+
+
+## Motivation
+
+Monitoring distributed systems [enables operators](https://landing.google.com/sre/sre-book/chapters/monitoring-distributed-systems/#why-monitor-pWsBTZ) to analyze long-term trends in system performance, compare experiments against a baseline, alert on failures, and conduct retrospective analysis on systems (debugging). These abilities are critical to keeping software correct and available.
+
+### User Stories
+
+* As a DCP operator, I want to compare experiments against a baseline, so that I can compare design alternatives.
+* As a DCP operator, I want to conduct retrospective analysis on systems, so that I can debug systems.
+* As a DCP operator, I want to analyze long-term trends in system performance, so that I can estimate changes in cost and performance.
+* As a DCP operator, I want to be alerted of system failure, so that I know when something goes wrong and can keep the system available for our users.
+
+## Detailed Design
+
+In summary, each system critical to user-facing features in DCP should have the following.
```

Comment:
What does 'user-facing' cover? If we assume that users are customers of the system rather than internal components, does this mean data consumers as in the GA doc or is this also intended to cover data contributors?

Reply:
Ultimately, this applies to both consumers and contributors as they are both end users. If your system is not yet released to users, then perhaps you don't need to have all monitoring done. I would argue that it would hurt development velocity not to have this in place, however.

Reply:
I agree - all components need to do this stuff. I'm just thinking in terms of prioritization against other work and not spending too much work putting monitoring into an immature system that will change. This might be a different situation for ingest, where there may be more need for evolution compared to other components.

Reply:
On the topic of this document, do you have any concrete suggestions here or should I resolve this thread?

Reply:
Could you clarify in the text that user-facing applies to both data submitters and consumers? I know this would be the normal interpretation but I just want to make sure there is a distinction here between the use of 'user-facing' in the GA doc where it's only being applied to data consumers.